vyper/builtin_functions/signatures.py

```diff
@@ -101,43 +46,23 @@ def validate_inputs(wrapped_fn):
 
     @functools.wraps(wrapped_fn)
     def decorator_fn(self, node, context):
-        argz = [i[1] for i in self._inputs]
+        arg_types = [a._metadata["type"] for a in node.args]
         kwargz = getattr(self, "_kwargs", {})
-        function_name = node.func.id
-        if len(node.args) > len(argz):
-            raise StructureException(
-                f"Expected {len(argz)} arguments for {function_name}, got {len(node.args)}", node
-            )
-        subs = []
-        for i, expected_arg in enumerate(argz):
-            if len(node.args) > i:
-                subs.append(
-                    process_arg(
-                        i + 1,
-                        node.args[i],
-                        expected_arg,
-                        function_name,
-                        context,
-                    )
-                )
-            elif isinstance(expected_arg, Optional):
-                subs.append(expected_arg.default)
-            else:
-                raise StructureException(f"Not enough arguments for function: {node.func.id}", node)
+        assert len(node.args) == len(arg_types)
+        subs = [process_arg(arg, arg_type, context) for arg, arg_type in zip(node.args, arg_types)]
         kwsubs = {}
         node_kw = {k.arg: k.value for k in node.keywords}
-        for k, expected_arg in kwargz.items():
-            if k not in node_kw:
-                if not isinstance(expected_arg, Optional):
-                    raise StructureException(
-                        f"Function {function_name} requires argument {k}", node
-                    )
-                kwsubs[k] = expected_arg.default
-            else:
-                kwsubs[k] = process_arg(k, node_kw[k], expected_arg, function_name, context)
-        for k, _arg in node_kw.items():
-            if k not in kwargz:
-                raise StructureException(f"Unexpected argument: {k}", node)
+        node_kw_types = {k.arg: k.value._metadata["type"] for k in node.keywords}
+        if kwargz:
+            for k, expected_arg in self._kwargs.items():
+                if k not in node_kw:
+                    kwsubs[k] = expected_arg.default
+                else:
+                    # For literals, skip process_arg and set the AST node value
+                    if expected_arg.require_literal:
+                        kwsubs[k] = node_kw[k].n
```

Comment:
why inline this behavior but all other branches get delegated down to `process_arg`?

Reply:
Since `require_literal` is an attribute of `Optional`, but the `Optional` object is not available in `process_arg`, it therefore needs to be handled at this stage.

This is meant to handle `max_outsize` specifically at the moment.